internal/devpkg/narinfo_cache.go

```diff
@@ -84,117 +86,118 @@ func FillNarInfoCache(ctx context.Context, packages ...*Package) error {
 	group, _ := errgroup.WithContext(ctx)
 	for _, p := range eligiblePackages {
 		pkg := p // copy the loop variable since its used in a closure below
-		names, err := pkg.GetOutputNames()
+		outputs, err := pkg.GetOutputs()
 		if err != nil {
 			return err
 		}
 
-		for _, o := range names {
-			output := o
+		for _, output := range outputs {
+			name := output.Name
 			group.Go(func() error {
-				_, err := pkg.fetchNarInfoStatusOnce(output)
+				_, err := pkg.fetchNarInfoStatusOnce(name)
 				return err
 			})
 		}
 	}
 	return group.Wait()
 }
 
-// narInfoStatusFnCache contains cached OnceValues functions that return cache
-// status for a package. In the future we can remove this cache by caching
-// package objects and ensuring packages are shared globally.
-var narInfoStatusFnCache = sync.Map{}
-
-// fetchNarInfoStatusOnce is like fetchNarInfoStatus, but will only ever run
-// once and cache the result.
-func (p *Package) fetchNarInfoStatusOnce(output string) (bool, error) {
-	type inCacheFunc func() (bool, error)
-	f, ok := narInfoStatusFnCache.Load(p.Raw)
-	if !ok {
-		f = inCacheFunc(sync.OnceValues(func() (bool, error) { return p.fetchNarInfoStatus(output) }))
-		f, _ = narInfoStatusFnCache.LoadOrStore(p.keyForOutput(output), f)
+// areExpectedOutputsInCacheOnce wraps fetchNarInfoStatusOnce and returns true
+// if the expected outputs are in the cache.
+func (p *Package) areExpectedOutputsInCacheOnce(
+	outputName string,
+) (bool, error) {
+	outputToCache, err := p.fetchNarInfoStatusOnce(outputName)
+	if err != nil {
+		return false, err
 	}
-	return f.(inCacheFunc)()
-}
-
-func (p *Package) keyForOutput(output string) string {
-	if output == useDefaultOutput {
-		sysInfo, err := p.sysInfoIfExists()
-		// let's be super safe to always avoid empty key.
-		if err == nil && sysInfo != nil && len(sysInfo.DefaultOutputs()) > 0 {
-			names := make([]string, len(sysInfo.DefaultOutputs()))
-			for i, o := range sysInfo.DefaultOutputs() {
-				names[i] = o.Name
-			}
-			slices.Sort(names)
-			output = strings.Join(names, ",")
+	if outputName == useDefaultOutputs {
```

Comment:
Not related to this PR, but `*` is also a valid output name to indicate all outputs. I don't think we're handling that right now. Does that mean devbox will just fallback to not using the cached store paths for `*` outputs (which I think is okay for now)?

Reply:
I'm not sure it will work at all because of how we handle ouputs (independent of cache). But good call, we should fix this in follow up.